File: daily/cpp/231_power_of_two.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
 public:
  bool isPowerOfTwo(int n) {
    if (n == 1) return true;
    if (n < 1) return false;

    while (n % 2 == 0 && n > 1) {
      n /= 2;
    }
    if (n == 1) return true;
    return false;
  }
};

File: daily/rust/231_power_of_two.rs
impl Solution {
    pub fn is_power_of_two(n: i32) -> bool {
        if n == 1 { return true; }
        if n < 1 { return false; }
        let mut m = n;
        while m % 2 == 0 && m > 1 {
            m = m / 2;
        }
        if m == 1 { return true; }
        false
    }
}
